flake.nix

```
@@ -25,18 +25,37 @@
           mesonBuildType = "release";
 
           outputs = [ "out" "dev" ];
-          outputBin = "dev";
 
           buildInputs = with pkgs; [ zlib ];
           nativeBuildInputs = with pkgs; [ pkg-config ninja meson ];
         };
         defaultPackage = packages.libspng;
 
-        # TODO checks
+        checks.libspng = pkgs.stdenv.mkDerivation {
+          pname = "libspng";
+          name = "libspng";
+          src = libspng-src;
+
+          doCheck = true;
+
+          mesonBuildType = "release";
+
+          mesonFlags = [
+            # this is required to enable testing
+            # https://github.com/randy408/libspng/blob/bc383951e9a6e04dbc0766f6737e873e0eedb40b/tests/README.md#testing
+            "-Ddev_build=true"
+          ];
+
+          outputs = [ "out" "dev" ];
+
+          checkInputs = with pkgs; [ cmake libpng ];
+          buildInputs = with pkgs; [ pkg-config zlib ];
+          nativeBuildInputs = with pkgs; [ ninja meson ];
+        };
 
         # `nix develop`
         devShell = pkgs.mkShell {
-          nativeBuildInputs = with pkgs; [ pkg-config zlib meson ninja ];
+          nativeBuildInputs = with pkgs; [ pkg-config zlib meson ninja cmake ];
```

You might be able to do a similar thing here, and just pull imports from the check defined above:

```suggestion
          inputsFrom = [ checks.libspng ];
```

Just like above, this would also help reduce duplication.

---

Oh, I didn't test that this actually gives you everything you need to compile the package.  You might want to do that.

(Although I did test that pkg-config, zlib, meson, ninja, cmake, and libpng were all available in the `nix develop` environment, so my guess is that it is fine.)